src/offline/cable_output.F90

```diff
@@ -1633,1266 +1651,526 @@ SUBROUTINE write_output(dels, ktau, met, canopy, casaflux, casapool, casamet, ss
 
-       out%LittCarbMetabolic = out%LittCarbMetabolic + REAL(casapool%clitter(:,1) / 1000.0, 4)
-       IF(writenow) THEN
-          ! Divide accumulated variable by number of accumulated time steps:
-          out%LittCarbMetabolic = out%LittCarbMetabolic / REAL(output%interval, 4)
-          ! Write value to file:
-          CALL write_ovar(out_timestep, ncid_out, ovid%LittCarbMetabolic, 'LittCarbMetabolic', out%LittCarbMetabolic,    &
-               ranges%TotLittCarb, patchout%LittCarbMetabolic, 'default', met)
-          ! Reset temporary output variable:
-          out%LittCarbMetabolic = 0.0
-       END IF
-
-       out%LittCarbStructural = out%LittCarbStructural + REAL(casapool%clitter(:,2) / 1000.0, 4)
-       IF(writenow) THEN
-          ! Divide accumulated variable by number of accumulated time steps:
-          out%LittCarbStructural = out%LittCarbStructural / REAL(output%interval, 4)
-          ! Write value to file:
-          CALL write_ovar(out_timestep, ncid_out, ovid%LittCarbStructural, 'LittCarbStructural', out%LittCarbStructural, &
-               ranges%TotLittCarb, patchout%LittCarbStructural, 'default', met)
-          ! Reset temporary output variable:
-          out%LittCarbStructural = 0.0
-       END IF
+     INTEGER :: ncid_file
 
-       out%LittCarbCWD = out%LittCarbCWD + REAL(casapool%clitter(:,3) / 1000.0, 4)
-       IF(writenow) THEN
-          ! Divide accumulated variable by number of accumulated time steps:
-          out%LittCarbCWD = out%LittCarbCWD / REAL(output%interval, 4)
+     CALL check_range(pname, out_par, prange, out_timestep, met)
+    
+     IF (output_par) THEN
+          IF (PRESENT(restart)) THEN
+               ncid_file = ncid_restart
+          ELSE
+               ncid_file = ncid_out
+          END IF
           ! Write value to file:
-          CALL write_ovar(out_timestep, ncid_out, ovid%LittCarbCWD, 'LittCarbCWD', out%LittCarbCWD,    &
-               ranges%TotLittCarb, patchout%LittCarbCWD, 'default', met)
-          ! Reset temporary output variable:
-          out%LittCarbCWD = 0.0
-       END IF
+          CALL write_ovar (ncid_file, parID, pname, out_par,       &
+               writepatch, dimswitch, restart)
+     END IF
 
+  END SUBROUTINE check_and_write_d2_p
 
-       out%PlantCarbLeaf = out%PlantCarbLeaf + REAL(casapool%cplant(:,1) / 1000.0, 4)
-       IF(writenow) THEN
-          ! Divide accumulated variable by number of accumulated time steps:
-          out%PlantCarbLeaf = out%PlantCarbLeaf / REAL(output%interval, 4)
-          ! Write value to file:
-          CALL write_ovar(out_timestep, ncid_out, ovid%PlantCarbLeaf, 'PlantCarbLeaf', out%PlantCarbLeaf,    &
-               ranges%TotLittCarb, patchout%PlantCarbLeaf, 'default', met)
-          ! Reset temporary output variable:
-          out%PlantCarbLeaf = 0.0
-       END IF
+  SUBROUTINE generate_out_write_acc_d1(output_var, varID, vname, out_var, acc_val, vrange, writepatch, met)
+     !! TODO: Remove met
```

Comment:
Why do you want to remove met? Don't you need it in `check_range`? Do you want to move to get it through a USE statement instead of an input arg?

Reply:
I was trying to minimize the list of parameters such as for `met`, `dimswitch`, `restart`. There are some aspects to this

- Another file landuse is calling `check_and_write` 
- We don't want circular dependencies w.r.t. import statements (Like `USE cable_driver_module` wouldn't be good since 
- State management of local variables could be tricky, so a functional style would be good here 

So I'll create a type for `output_settings` and add it as a parameter (I was thinking of doing that hence the TODO)